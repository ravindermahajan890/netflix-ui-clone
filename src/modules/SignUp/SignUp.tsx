import LockOutlinedIcon from '@mui/icons-material/LockOutlined';
import PersonOutlineIcon from '@mui/icons-material/PersonOutline';
import { Box, Card, CardHeader, Container } from '@mui/material';
import { useState } from 'react';
import { useNavigate } from 'react-router-dom';
import CustomButton from '../../components/CustomButton';
import Input from '../../components/Input';
import Logo from '../../components/Logo';
import { useAuth } from '../../providers/Auth/Auth';

const SignUp = () => {
  const [email, setEmail] = useState<string>('');
  const [password, setPassword] = useState<string>('');
  const [confirmPassword, setConfirmPassword] = useState<string>('');

  const { signUp } = useAuth();
  const navigate = useNavigate();

  const onSubmitHandler = async () => {
    if (password !== confirmPassword) {
      console.log('[SignUp] Confirm password does not match');
      return;
    }

    await signUp(email, password);

    navigate('/', { replace: true });
  };

  return (
    <Box display="flex" justifyContent="center" alignItems="center" minHeight="100vh">
      <Container maxWidth="sm">
        <Card style={{ padding: '24px 64px 40px' }}>
          <CardHeader title={<Logo alignment="center" />} />
          <Box display="flex" flexDirection="column" gap="30px" marginTop="16px">
            <Input
              label="Email"
              icon={<PersonOutlineIcon />}
              type="email"
              onChange={(event) => setEmail((event.target as HTMLInputElement).value)}
            />
            <Input
              label="Password"
              icon={<LockOutlinedIcon />}
              type="password"
              onChange={(event) => setPassword((event.target as HTMLInputElement).value)}
            />
            <Input
              label="Confirm password"
              icon={<LockOutlinedIcon />}
              type="password"
              onChange={(event) => setConfirmPassword((event.target as HTMLInputElement).value)}
            />
            <CustomButton variant="contained" color="tertiary" size="large" onClick={onSubmitHandler}>
              Signup
            </CustomButton>
          </Box>
        </Card>
      </Container>
    </Box>
  );
};

export default SignUp;
