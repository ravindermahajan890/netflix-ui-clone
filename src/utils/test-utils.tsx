// eslint-disable-next-line import/no-extraneous-dependencies
import { render as rtlRender } from '@testing-library/react';
import { PropsWithChildren, ReactElement } from 'react';
import { HashRouter } from 'react-router-dom';
import Providers from '../providers';

// eslint-disable-next-line import/prefer-default-export
export const render = (ui: ReactElement) => {
  const wrapper = ({ children }: PropsWithChildren) => (
    <Providers>
      <HashRouter>{children}</HashRouter>
    </Providers>
  );

  return rtlRender(ui, { wrapper });
};
