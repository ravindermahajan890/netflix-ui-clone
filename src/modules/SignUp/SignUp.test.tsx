import { render } from '../../utils/test-utils';
import SignUp from '.';

describe('CustomButton', () => {
  it('should render', () => {
    const { asFragment } = render(<SignUp />);

    expect(asFragment()).toMatchSnapshot();
  });
});
