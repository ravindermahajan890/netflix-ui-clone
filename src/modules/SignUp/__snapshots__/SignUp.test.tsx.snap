// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomButton should render 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  min-height: 100vh;
}

.emotion-1 {
  width: 100%;
  margin-left: auto;
  box-sizing: border-box;
  margin-right: auto;
  display: block;
  padding-left: 16px;
  padding-right: 16px;
}

@media (min-width:600px) {
  .emotion-1 {
    padding-left: 24px;
    padding-right: 24px;
  }
}

@media (min-width:600px) {
  .emotion-1 {
    max-width: 600px;
  }
}

.emotion-2 {
  background-color: #000;
  color: #fff;
  -webkit-transition: box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  border-radius: 4px;
  box-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12);
  background-image: linear-gradient(rgba(255, 255, 255, 0.05), rgba(255, 255, 255, 0.05));
  overflow: hidden;
}

.emotion-3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 16px;
}

.emotion-4 {
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
}

.emotion-5 {
  margin: 0;
  font-family: Inter;
  font-weight: 400;
  font-size: 1.5rem;
  line-height: 1.334;
  display: block;
}

.emotion-6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 12px;
}

.emotion-7 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  width: 40px;
  height: 40px;
  font-family: Inter;
  font-size: 1.25rem;
  line-height: 1;
  border-radius: 50%;
  overflow: hidden;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.emotion-8 {
  width: 100%;
  height: 100%;
  text-align: center;
  object-fit: cover;
  color: transparent;
  text-indent: 10000px;
}

.emotion-9 {
  margin: 0;
  font-family: Inter;
  font-weight: 400;
  font-size: 1.5rem;
  line-height: 1.334;
  color: #FE0039;
}

.emotion-10 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 30px;
  margin-top: 16px;
}

.emotion-11 {
  color: rgba(255, 255, 255, 0.7);
  font-family: Inter;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.4375em;
  padding: 0;
  position: relative;
  display: block;
  transform-origin: top left;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
  max-width: 100%;
  -webkit-transition: color 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms,-webkit-transform 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms,max-width 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
  transition: color 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms,transform 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms,max-width 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
  color: white;
  font-size: 20px;
}

.emotion-11.Mui-focused {
  color: #000;
}

.emotion-11.Mui-disabled {
  color: rgba(255, 255, 255, 0.5);
}

.emotion-11.Mui-error {
  color: #DC3545;
}

.emotion-12 {
  font-family: Inter;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.4375em;
  color: #fff;
  box-sizing: border-box;
  position: relative;
  cursor: text;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border: 1px solid #ADB5BD;
  border-radius: 4px;
  width: 100%;
  padding: 16px;
  margin-top: 8px;
}

.emotion-12.Mui-disabled {
  color: rgba(255, 255, 255, 0.5);
  cursor: default;
}

.emotion-12 .MuiInputBase-input {
  font-size: 20px;
  font-weight: 300;
  padding: 0;
}

.emotion-13 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 0.01em;
  max-height: 2em;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  white-space: nowrap;
  color: #fff;
  margin-right: 8px;
}

.emotion-14 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: 1em;
  height: 1em;
  display: inline-block;
  fill: currentColor;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  font-size: 1.5rem;
}

.emotion-15 {
  font: inherit;
  letter-spacing: inherit;
  color: currentColor;
  padding: 4px 0 5px;
  border: 0;
  box-sizing: content-box;
  background: none;
  height: 1.4375em;
  margin: 0;
  -webkit-tap-highlight-color: transparent;
  display: block;
  min-width: 0;
  width: 100%;
  -webkit-animation-name: mui-auto-fill-cancel;
  animation-name: mui-auto-fill-cancel;
  -webkit-animation-duration: 10ms;
  animation-duration: 10ms;
}

.emotion-15::-webkit-input-placeholder {
  color: currentColor;
  opacity: 0.5;
  -webkit-transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
}

.emotion-15::-moz-placeholder {
  color: currentColor;
  opacity: 0.5;
  -webkit-transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
}

.emotion-15:-ms-input-placeholder {
  color: currentColor;
  opacity: 0.5;
  -webkit-transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
}

.emotion-15::-ms-input-placeholder {
  color: currentColor;
  opacity: 0.5;
  -webkit-transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
}

.emotion-15:focus {
  outline: 0;
}

.emotion-15:invalid {
  box-shadow: none;
}

.emotion-15::-webkit-search-decoration {
  -webkit-appearance: none;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-15::-webkit-input-placeholder {
  opacity: 0!important;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-15::-moz-placeholder {
  opacity: 0!important;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-15:-ms-input-placeholder {
  opacity: 0!important;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-15::-ms-input-placeholder {
  opacity: 0!important;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-15:focus::-webkit-input-placeholder {
  opacity: 0.5;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-15:focus::-moz-placeholder {
  opacity: 0.5;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-15:focus:-ms-input-placeholder {
  opacity: 0.5;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-15:focus::-ms-input-placeholder {
  opacity: 0.5;
}

.emotion-15.Mui-disabled {
  opacity: 1;
  -webkit-text-fill-color: rgba(255, 255, 255, 0.5);
}

.emotion-15:-webkit-autofill {
  -webkit-animation-duration: 5000s;
  animation-duration: 5000s;
  -webkit-animation-name: mui-auto-fill;
  animation-name: mui-auto-fill;
}

.emotion-26 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  font-family: Inter;
  font-weight: 500;
  font-size: 0.9375rem;
  line-height: 1.75;
  text-transform: uppercase;
  min-width: 64px;
  padding: 8px 22px;
  border-radius: 4px;
  -webkit-transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  background-color: #842029;
  box-shadow: 0px 3px 1px -2px rgba(0,0,0,0.2),0px 2px 2px 0px rgba(0,0,0,0.14),0px 1px 5px 0px rgba(0,0,0,0.12);
  text-transform: none;
  font-size: 20px;
  font-weight: 400;
}

.emotion-26::-moz-focus-inner {
  border-style: none;
}

.emotion-26.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

@media print {
  .emotion-26 {
    -webkit-print-color-adjust: exact;
    color-adjust: exact;
  }
}

.emotion-26:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
  box-shadow: 0px 2px 4px -1px rgba(0,0,0,0.2),0px 4px 5px 0px rgba(0,0,0,0.14),0px 1px 10px 0px rgba(0,0,0,0.12);
}

@media (hover: none) {
  .emotion-26:hover {
    background-color: #842029;
  }
}

.emotion-26:active {
  box-shadow: 0px 5px 5px -3px rgba(0,0,0,0.2),0px 8px 10px 1px rgba(0,0,0,0.14),0px 3px 14px 2px rgba(0,0,0,0.12);
}

.emotion-26.Mui-focusVisible {
  box-shadow: 0px 3px 5px -1px rgba(0,0,0,0.2),0px 6px 10px 0px rgba(0,0,0,0.14),0px 1px 18px 0px rgba(0,0,0,0.12);
}

.emotion-26.Mui-disabled {
  color: rgba(255, 255, 255, 0.3);
  box-shadow: none;
  background-color: rgba(255, 255, 255, 0.12);
}

.emotion-27 {
  overflow: hidden;
  pointer-events: none;
  position: absolute;
  z-index: 0;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  border-radius: inherit;
}

<div
    class="MuiBox-root emotion-0"
  >
    <div
      class="MuiContainer-root MuiContainer-maxWidthSm emotion-1"
    >
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root emotion-2"
        style="padding: 24px 64px 40px;"
      >
        <div
          class="MuiCardHeader-root emotion-3"
        >
          <div
            class="MuiCardHeader-content emotion-4"
          >
            <span
              class="MuiTypography-root MuiTypography-h5 MuiCardHeader-title emotion-5"
            >
              <div
                class="MuiBox-root emotion-6"
              >
                <div
                  class="MuiAvatar-root MuiAvatar-circular emotion-7"
                >
                  <img
                    alt="TAMASHFLIX"
                    class="MuiAvatar-img emotion-8"
                    src="/tamashflix.svg"
                  />
                </div>
                <h5
                  class="MuiTypography-root MuiTypography-h5 emotion-9"
                >
                  TAMASHFLIX
                </h5>
              </div>
            </span>
          </div>
        </div>
        <div
          class="MuiBox-root emotion-10"
        >
          <div>
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-animated MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-animated emotion-11"
            >
              Email
            </label>
            <div
              class="MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-adornedStart emotion-12"
            >
              <div
                class="MuiInputAdornment-root MuiInputAdornment-positionStart emotion-13"
              >
                <span
                  class="notranslate"
                >
                  ​
                </span>
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium emotion-14"
                  data-testid="PersonOutlineIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 5.9c1.16 0 2.1.94 2.1 2.1s-.94 2.1-2.1 2.1S9.9 9.16 9.9 8s.94-2.1 2.1-2.1m0 9c2.97 0 6.1 1.46 6.1 2.1v1.1H5.9V17c0-.64 3.13-2.1 6.1-2.1M12 4C9.79 4 8 5.79 8 8s1.79 4 4 4 4-1.79 4-4-1.79-4-4-4zm0 9c-2.67 0-8 1.34-8 4v3h16v-3c0-2.66-5.33-4-8-4z"
                  />
                </svg>
              </div>
              <input
                class="MuiInputBase-input MuiInputBase-inputAdornedStart emotion-15"
                type="email"
                value=""
              />
            </div>
          </div>
          <div>
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-animated MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-animated emotion-11"
            >
              Password
            </label>
            <div
              class="MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-adornedStart emotion-12"
            >
              <div
                class="MuiInputAdornment-root MuiInputAdornment-positionStart emotion-13"
              >
                <span
                  class="notranslate"
                >
                  ​
                </span>
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium emotion-14"
                  data-testid="LockOutlinedIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M18 8h-1V6c0-2.76-2.24-5-5-5S7 3.24 7 6v2H6c-1.1 0-2 .9-2 2v10c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V10c0-1.1-.9-2-2-2zM9 6c0-1.66 1.34-3 3-3s3 1.34 3 3v2H9V6zm9 14H6V10h12v10zm-6-3c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2z"
                  />
                </svg>
              </div>
              <input
                class="MuiInputBase-input MuiInputBase-inputAdornedStart emotion-15"
                type="password"
                value=""
              />
            </div>
          </div>
          <div>
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-animated MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-animated emotion-11"
            >
              Confirm password
            </label>
            <div
              class="MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-adornedStart emotion-12"
            >
              <div
                class="MuiInputAdornment-root MuiInputAdornment-positionStart emotion-13"
              >
                <span
                  class="notranslate"
                >
                  ​
                </span>
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium emotion-14"
                  data-testid="LockOutlinedIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M18 8h-1V6c0-2.76-2.24-5-5-5S7 3.24 7 6v2H6c-1.1 0-2 .9-2 2v10c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V10c0-1.1-.9-2-2-2zM9 6c0-1.66 1.34-3 3-3s3 1.34 3 3v2H9V6zm9 14H6V10h12v10zm-6-3c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2z"
                  />
                </svg>
              </div>
              <input
                class="MuiInputBase-input MuiInputBase-inputAdornedStart emotion-15"
                type="password"
                value=""
              />
            </div>
          </div>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedTertiary MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButton-root MuiButton-contained MuiButton-containedTertiary MuiButton-sizeLarge MuiButton-containedSizeLarge emotion-26"
            tabindex="0"
            type="button"
          >
            Signup
            <span
              class="MuiTouchRipple-root emotion-27"
            />
          </button>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
